### ISSUE TITLE
Releases

### ISSUE THREAD

**Issue by pedromorgan** [ts2/ts2#38]
Can we touch on how to build and release ts2..

Options are..
a pip install..

And exe..
Now exe Am unsure about, for windows.. and we will need win32/win64
and for fun we can throw in ARM = raspberry pi..

So can @npiganeau explain current build sceneraio, how its done etc... ?

In the back of my mind, its quite easy to set up a Jenkins instance on rhcloud..

**Comment by npiganeau** [comment]
Today ts2 is released as exe so that the massive amount of windows users can access ts2 as easily as possible. I only do x64 release because that is what I have on my PC...

This is a 2 step process:
1. Build the .exe with cx_freeze (this is the build.py script at the root of the source file). Just run
```python build.py build```. This will create the exe and dependencies (DLL, resources, etc.) in the build directory.
2. Create the installer with ISS using the setup.iss source file. The installer will be in the dist directory.

I wish to keep .exe release for the reason mentioned above, but we can consider any other release mechanism.

As for using Jenkins, why not, but if we want it to be worth it, we should write tests for ts2, and that's a pretty long job to do it seriously !

**Comment by pedromorgan** [comment]
Ok am assigning this to myself, for jenkins build server..

WE need slaves though..

I suppose we need...
- win binaries 32/64
- and NIX ?
 -- maybe pip install
 -- or even git install... for hardcore...

**Comment by npiganeau** [comment]
pip install is cool

Even better would be also to have DEBs and RPMs for major distributions (I believe that most of our users aren't hackers at all)

**Comment by pedromorgan** [comment]
Emm I have reservarions about deb and RPM..
mostly because of maintanace hell and forking and a crashed version not working a year later..

So for now if we stick to pip.. (but never done a proper pip install);..

**Comment by pedromorgan** [comment]
Have created and account in the "cheeseshop" for `pip install`

This needs a setup.py file inroot of project.. is @npiganeau samiliar with this...

Also I kinda want to stick the "version" into into a single `ts2/version.py` of alike.. so these can be loaded without all the pyqt libs and dependancies,, eg in sceneario such as a docs build where no libs are installed..

**Comment by npiganeau** [comment]
setup.py is a distutils/setuptools file. We already got one for cx_freeze which is an extension to distutils. We'll just need to add an 'install' target.

For version, up to now I put the info in utils.py by I saw you also put some metadat in ts2/__init__.py

Putting all metadata info, including version in ts2/__init__.py is fine for me, let's do that !

**Comment by npiganeau** [comment]
@pedromorgan I'm ok on my side to start the release process for ts2 v0.6. 

I've coded all the features I wanted in the [Roadmap](https://github.com/ts2/ts2/wiki/Roadmap), except:
- Train joigning, but that implies many changes that I would rather do after the [Next Arch](https://github.com/ts2/ts2/wiki/New-Arch) rewrite, so I postponed to v0.7
- UK shunt signals: but that's ts2-data now, so we can release ts2 and do it whenever we want.

This is the way I see the process:
1. You make PRs for things you want to add to v0.6 before feature freezing
2. We freeze the code and start testing to check that we have no regressions ( @Sharpe49 and @mcgiwer, your help is welcome !)
3. In the meantime, we update all the setup and release scripts, including for PyPI.
4. When we are ready, we fire v0.6 !

**Comment by npiganeau** [comment]
Oops I forgot: we need to update translations also during the freezing phase (French and Polish and maybe Welsh :-))

**Comment by mcgiwer** [comment]
@pedromorgan As I wrote in other post, the JSon files in the ts-data need a clean-up and standarisation. For now, they are so mixed up that it's quite hard to work with them...

@npiganeau I need a base to do the current translation work. Please update the ts2_pl.ts file by remove the unused anymore fields and add the ones needed for translation

**Comment by npiganeau** [comment]
translation files updated

**Comment by npiganeau** [comment]
French translation updated

**Comment by npiganeau** [comment]
It's done ! https://github.com/ts2/ts2/releases/tag/v0.6

**Comment by pedromorgan** [comment]
No t here.. seems like its not in master..

**Comment by npiganeau** [comment]
Did I forget to push ?

**Comment by npiganeau** [comment]
;-) Now it is !

**Comment by pedromorgan** [comment]
So  remove v0.6 staf and make 0.6.1 tag ..

Works here.. on master.. so not a disaster..
Some kinda poem in there..

**Comment by npiganeau** [comment]
Yep ! I did a 0.6.0 instead. So now should be ok !

**Comment by npiganeau** [comment]
Now we need advertising on ts2.github.io !

**Comment by pedromorgan** [comment]
Yep into that.. and soem go devs, arduino freaks, graphic designers.. ;-))
 He He the future as a FOSS project..

I also want to automate a bit.. so new "tag" would be a new release auto magically ;-)

**Comment by npiganeau** [comment]
Ok

**Comment by pedromorgan** [comment]
irc.freenode.com#trainsigsim would be handy sometimes

**Comment by pedromorgan** [comment]
How do meake this one.. IIS and wind
with wndows
https://github.com/ts2/ts2/blob/master/setup.iss

**Comment by pedromorgan** [comment]
anyway.. gettign to grips with py build, as I uipdate apps from github..

**Comment by npiganeau** [comment]
I've put it all in the build.py. Just run ```python build.py build``` and it will make you an exe installer for an exe freeze of TS2. 

Didn't try the script on other OS, but freezing really makes sense on windows only.

**Comment by pedromorgan** [comment]
I would like specific "awarness" of what release is moving forwards..

I now neeed to update www etc.. and pre warned would be handy..eg 24 hours notice so I can make soem noise.

**Comment by npiganeau** [comment]
Yep, sorry, I was too impatient I suppose ...

**Comment by pedromorgan** [comment]
http://ts2.github.io/ is updated manually..

Am trying to make this kinda automated so when "new tag" is created..
then system auto updated website et all

**Comment by pedromorgan** [comment]
and if @npiganeau enters IRC then ping me..
